src/server/config.rs

```diff
@@ -185,12 +191,30 @@ impl Config {
                 self.advertise_addr
             ));
         }
+        if self.status_addr.is_empty() && !self.advertise_status_addr.is_empty() {
+            return Err(box_err!("status-addr can not be empty"));
+        }
         if !self.status_addr.is_empty() {
             box_try!(config::check_addr(&self.status_addr));
+            if !self.advertise_status_addr.is_empty() {
+                box_try!(config::check_addr(&self.advertise_status_addr));
+            } else {
+                info!(
+                    "no advertise-status-addr is specified, falling back to status-addr";
+                    "status-addr" => %self.status_addr
+                );
+                self.advertise_status_addr = self.status_addr.clone();
+            }
+            if self.advertise_status_addr.starts_with("0.0.0.0") {
```

Comment:
What if status_add is `0.0.0.0` and adverties_status_addr is empty? Will it report error `invalid advertise-status-addr`?

Reply:
Yes, if `adverties_status_addr` is empty, it will fall back to `status_add ` and report the `invalid advertise-status-addr` error.